llvm/test/lab3/sharapov_georgiy/test.mir

```diff
@@ -0,0 +1,378 @@
+# RUN: llc -mtriple x86_64-unknown-linux-gnu --load=%llvmshlibdir/X86MulattoPass%shlibext -run-pass=x86-mulatto-pass %s -o - | FileCheck %s
+
+# clang -O2 -emit-llvm -S 1.c -o 2_1.ll
+# llc -simplify-mir -march=x86-64 2_1.ll -stop-after=x86-isel -o test.mir
+
+
+# #include <immintrin.h>
+
+# // 1
+
+# __m128d basic_opt(__m128d a1, __m128d a2, __m128d a3, __m128d a4) {        
+#     __m128d res = _mm_setzero_pd();
+#     
+#     __m128d tmp = a1 * a2;
+#     res = tmp + a3;
+#     
+#     return res;
+# }
+
+# // 2
+
+# __m128d no_opt_read(__m128d a1, __m128d a2, __m128d a3, __m128d a4) {        
+#     __m128d res = _mm_setzero_pd();
+# 
+#     __m128d tmp = a1 * a2;
+# 
+#     __m128d smth = tmp / a3; 
+# 
+#     res = tmp + a3;
+#     
+#     return res + smth;
+# }
+
+# // 3
+
+# __m128d no_opt_changed(__m128d a1, __m128d a2, __m128d a3, __m128d a4) {        
+#     __m128d res = _mm_setzero_pd();
+# 
+#     __m128d tmp = a1 * a2;
+# 
+#     tmp = a1 + a2;
+# 
+#     res = tmp + a3;
+#     
+#     return res + tmp;
+# }
+
+# // 4
+
+# __m128d multiple_opt(__m128d a1, __m128d a2, __m128d a3, __m128d a4) {        
+#     __m128d tmp1 = _mm_setzero_pd();
+#     __m128d tmp2 = _mm_setzero_pd();
+#     __m128d tmp3 = _mm_setzero_pd();
+#     __m128d res1 = _mm_setzero_pd();
+#     __m128d res2 = _mm_setzero_pd();
+#     __m128d res3 = _mm_setzero_pd();
+# 
+#     tmp1 = a2 * a3; 
+#     tmp2 = a1 * a2; 
+# 
+#     res1 = tmp1 + a1;
+#     res2 = tmp2 + a2;
+# 
+#     tmp3 = a3 * a1; 
+#     res3 = tmp3 + a3;
+# 
+#     return res1 + res2 + res3;
+# }
+
+# // 5
+
+# __m128d no_opt_and_mult_opt(__m128d a1, __m128d a2, __m128d a3, __m128d a4) {        
+#     __m128d tmp1 = _mm_setzero_pd();
+#     __m128d tmp2 = _mm_setzero_pd();
+#     __m128d tmp3 = _mm_setzero_pd();
+#     __m128d res1 = _mm_setzero_pd();
+#     __m128d res2 = _mm_setzero_pd();
+#     __m128d res3 = _mm_setzero_pd();
+# 
+#     tmp1 = a2 * a3; 
+#     tmp2 = a1 * a2; 
+# 
+#     tmp3 = a3 * a1; 
+#     __m128d smth = tmp3 / a3; 
+#     res3 = tmp3 + a3;
```

Comment:
Could you please check the following modified case?
```suggestion
#     res3 = tmp3 + a3;
#     __m128d smth = tmp3 / a3; 
```

Reply:
I added the suggested test and made changes in the pass for such cases.